### PR TITLE
Add accessor list function to token store

### PR DESCRIPTION
**Pull request by jefferai**

### CODE REVIEW

File: vault/token_store.go [hunk 1]
```diff
@@ -563,7 +563,17 @@ func (ts *TokenStore) createAccessor(entry *TokenEntry) error {
...
 	path := accessorPrefix + ts.SaltID(entry.Accessor)
-	le := &logical.StorageEntry{Key: path, Value: []byte(entry.ID)}
+
+	aEntry := &accessorEntry{
+		TokenID:    entry.ID,
+		AccessorID: entry.Accessor,
+	}
+	aEntryBytes, err := jsonutil.EncodeJSON(aEntry)
+	if err != nil {
+		return fmt.Errorf("failed to marshel accessor index entry: %v", err)
```
Comment (LLBennett) [inline vault/token_store.go]: s/marshel/marshal

**Comment by LLBennett** [comment]
LGTM.

File: vault/token_store.go [hunk 2]
```diff
@@ -845,16 +894,33 @@ func (ts *TokenStore) handleCreateAgainstRole(
...
+
+	err = jsonutil.DecodeJSON(entry.Value, &aEntry)
+	// If we hit an error, assume it's a pre-struct straight token ID
+	if err != nil {
+		aEntry.TokenID = string(entry.Value)
+		te, err := ts.lookupSalted(ts.SaltID(aEntry.TokenID))
+		if err != nil {
+			return accessorEntry{}, fmt.Errorf("failed to look up token using accessor index: %s", err)
+		}
+		aEntry.AccessorID = te.Accessor
```
Comment (vishalnayak) [inline vault/token_store.go]: Looks like AccessorID from this storage entry is not used anywhere.
Can this be removed?

Comment (jefferai) [inline vault/token_store.go]: Look again!   :-D

Comment (vishalnayak) [inline vault/token_store.go]: OMG. I see it now :)
It is *the* thing required.

Comment (jefferai) [inline vault/token_store.go]: Haha, it'd be odd of me to explicitly add that in to not use it :-)

**Comment by vishalnayak** [comment]
LGTM!

**Pull request merged**